packages/checkout/filter-registry/index.ts

```diff
@@ -107,9 +199,16 @@ export const __experimentalApplyCheckoutFilter = < T >( {
 	/** Function that needs to return true when the filtered value is passed in order for the filter to be applied. */
 	validation?: ( value: T ) => true | Error;
 } ): T => {
+	const cachedValues = useRef< Record< string, T > >( {} );
```

Comment:
Any reason why `cachedValues` is a ref and not a global variable?

Reply:
No specific reason, feel free to change it to a global variable.